recipe/meta.yaml

```
@@ -18,35 +18,49 @@ build:
     - numba = numba.misc.numba_entry:main
   script:
     - {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv
+  missing_dso_whitelist: # [osx]
+    # optional dependency: required only when omp is chosen as the backend for
+    # the threading layer
+    - lib/libiomp5.dylib # [osx]
   ignore_run_exports:
-    # TBB is entirely optional
-    - tbb
+    # tbb-devel triggers hard dependency on tbb, this is not the case.
+    - tbb # [not (armv6l or armv7l or aarch64 or linux32)]
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+    # both of these are needed on osx, llvm for the headers, Intel for the lib
+    - llvm-openmp              # [osx]
+    - intel-openmp             # [osx]
 
   host:
     - python
     - pip
     - setuptools
-    - llvmlite 0.34.0
+    - llvmlite 0.35.*
     - numpy
-    - tbb-devel >=2019.5  # [linux64 or ppc64le]
+    - tbb-devel >=2019.5,<=2020.3  # [linux64 or ppc64le]
 
   run:
     - python
-    - llvmlite >=0.34.0,<0.35
+    - llvmlite 0.35.*
```

Maybe we should switch to just using `pin_compatible` here. This is really what it is designed for.

```suggestion
    - {{ pin_compatible('llvmlite', max_pin='x.x') }}
```

---

That sounds right - how does it pick up the 0.35 here, though? Also, can't llvmlite broadcast that it needs this level of pinning (making the max_pin unneeded)? Not very familiar with using pin_compatible on something that's not numpy. :)

---

It picks it up based on the value in `host`.

Yeah we could add `run_exports` to `llvmlite`. Happy to add that. Though we probably shouldn't rely on it until the next `llvmlite`/`numba` release.

`pin_compatible` works the same for other packages. We use it with `numpy` as we do not want to add `run_exports` there as it would also affect pure Python packages that use `numpy`, which doesn't make much sense.